crates/bitwarden/src/platform/fido2/authenticator.rs

```diff
@@ -232,12 +295,24 @@ impl passkey::authenticator::CredentialStore for CredentialStoreImpl<'_> {
         ids: Option<&[passkey::types::webauthn::PublicKeyCredentialDescriptor]>,
         rp_id: &str,
     ) -> Result<Vec<Self::PasskeyItem>, StatusCode> {
+        #[derive(Debug, Error)]
+        enum InnerError {
+            #[error(transparent)]
+            VaultLocked(#[from] VaultLocked),
+            #[error(transparent)]
+            CipherError(#[from] CipherError),
+            #[error(transparent)]
+            CryptoError(#[from] CryptoError),
+            #[error(transparent)]
+            Fido2CallbackError(#[from] Fido2CallbackError),
+        }
+
         // This is just a wrapper around the actual implementation to allow for ? error handling
         async fn inner(
```

Comment:
I initially went with this inner function returning a Result to try to make error handling and conversion to StatusCode easier.  I wonder if it would be easier to just remove it and handle all the errors individually now, that way we won't need the innererror type

Reply:
Hmm, yea I wasn't sure what StatusCode was used for and I didn't want to affect the current behaviour.